src/controller/java/src/chip/devicecontroller/ChipDeviceController.java

```diff
@@ -314,6 +351,23 @@ public void onError(Throwable error) {
     completionListener.onError(error);
   }
 
+  public void onNOCChainGenerationNeeded(
```

Comment:
As a consideration, you may want to consider a wrapper class here to reduce the parameter count and also make it less likely to mix up the parameters (since they are all of type byte[]). e.g.

```
class NOCChainParams {
  byte[] getCsrElements()
  byte[] getCsrNonce()
}
```

etc.

Reply:
I will consider this for a followup PR, thanks.